flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointExecuteResult.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+/**
+ * The result of executing a checkpoint.
+ */
+public class CheckpointExecuteResult {
+
+	private final CompletedCheckpoint completedCheckpoint;
+
+	private final CheckpointAbortReason failure;
+
+	private final Throwable failureCause;
+
+	CheckpointExecuteResult(CompletedCheckpoint success) {
+		this.completedCheckpoint = success;
+		this.failure = null;
+		this.failureCause = null;
+	}
+
+	CheckpointExecuteResult(CheckpointAbortReason failure, Throwable failureCause) {
+		this.failure = failure;
+		this.failureCause = failureCause;
+		this.completedCheckpoint = null;
+	}
+
+	public boolean isSuccess() {
+		return completedCheckpoint != null;
+	}
+
+	public boolean isFailure() {
+		return failure != null;
+	}
+
+	public CompletedCheckpoint getCompletedCheckpoint() {
+		if (failure != null) {
+			throw new IllegalStateException("Checkpoint executing was failure"
+				+ (failureCause != null ? failureCause.getMessage() : "") + ".");
+		}
+		return completedCheckpoint;
+	}
+
+	public CheckpointAbortReason getFailureReason() {
+		if (failure != null) {
+			return failure;
+		} else {
+			throw new IllegalStateException("Checkpoint executing was successful");
+		}
+	}
+
+	public Throwable getFailureCause() {
+		if (getFailureReason() != null) {
+			return failureCause;
+		} else {
+			throw new IllegalStateException("Checkpoint executing was successful");
+		}
+	}
+
+	@Override
+	public String toString() {
+		return "CheckpointExecuteResult(" +
+			(isSuccess() ?
+				("completedCheckpoint: " + completedCheckpoint) :
+				("failure: " + failureCause == null ? "" : failureCause.getMessage())
```

Comment:
I think this requires parentheses to be correct: `(failureCause == null ? "" : failureCause.getMessage()))`